Review of MOM section in 25km config doc

dougiesquire opened the pull request:
Contributes to https://github.com/ACCESS-NRI/access-om3-configs/issues/904

--------------------


[Current rendered version here](https://access-om3-configs.access-hive.org.au/pr-previews/911/configurations/MC_25km/).

dougiesquire commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -4,90 +4,84 @@ The sections that follow explain why we selected each model parameter for the gl
 
-### Surface freezing and salinity constraints
-At the ocean surface, we've turned on frazil ice formation (`FRAZIL = TRUE`), which works upwards through each water column, transferring heat downwards from the layer above as needed to prevent the in-situ temperature falling below the local freezing point in each layer in turn. If the top layer is below freezing, heat is extracted from the sea ice model, which grows frazil ice in response. More ...
+The freezing conservative temperature is calculated from absolute salinity and pressure using a 23-term polynomial fit refactored from the TEOS-10 package (`TFREEZE_FORM = "TEOS_POLY"`). More relevant discussion can be found in [COSIMA TWG 23-July-2025 meeting minutes](https://forum.access-hive.org.au/t/cosima-twg-meeting-minutes-2025/4067/17#:~:text=Freezing%20temperature%20consistency%20between ...
```
```suggestion
The freezing conservative temperature is calculated from absolute salinity and pressure using a 23-term polynomial fit refactored from the TEOS-10 package (`TFREEZE_FORM = "TEOS_POLY"`). More relevant discussion can be found in the [COSIMA TWG 23-July-2025 meeting minutes](https://forum.access-hive.org.au/t/cosima-twg-meeting-minutes-2025/4067/17#:~:text=Freezing%20temperature%20consistency%20between%20mom6%20and%20cice). 
```

dougiesquire commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -4,90 +4,84 @@ The sections that follow explain why we selected each model parameter for the gl
 
 ### Surface salinity restoring
-Sea surface salinity is restored toward a reference climatology by enabling `RESTORE_SALINITY = True`. The restoration uses a monthly climatological dataset from the World Ocean Atlas 2023 (SALT_RESTORE_FILE = "salt_sfc_restore.nc"), available at [NOAA NCEI](https://www.ncei.noaa.gov/products/world-ocean-atlas). A piston velocity of 0.11 $m/day$ (`FLUXCONST = 0.11`) is applied to control the stre ...
+Sea surface salinity is restored toward a monthly climatological dataset calculated from the World Ocean Atlas 2023 available at [NOAA NCEI](https://www.ncei.noaa.gov/products/world-ocean-atlas) (`RESTORE_SALINITY = True`, `SALT_RESTORE_FILE = "salt_sfc_restore.nc"`). A piston velocity of 0.11 $m/day$ is applied to control the strength of the salinity relaxation (`FLUXCONST = 0.11`). The restorin ...
```
```suggestion
Sea surface salinity is restored toward a monthly climatological dataset calculated from the World Ocean Atlas 2023 available at [NOAA NCEI](https://www.ncei.noaa.gov/products/world-ocean-atlas) (`RESTORE_SALINITY = True`, `SALT_RESTORE_FILE = "salt_sfc_restore.nc"`). A piston velocity of 0.11 $m/day$ is applied to control the strength of the salinity relaxation (`FLUXCONST = 0.11`). The restoring is implemented as a virtual salt flux (`SRESTORE_AS_SFLUX = True`). This approach conserves salt overall, balanced globally by subtracting the mean flux to avoid altering global salinity (`ADJUST_NET_SRESTORE_TO_ZERO = True`). No effective limit is applied to the salinity restoring flux (`MAX_DELTA_SRESTORE = 999`). More discussion can be found in [issues/350](https://github.com/ACCESS-NRI/access-om3-configs/issues/350), [issues/325](https://github.com/ACCESS-NRI/access-om3-configs/issues/325), [issues/257](https://github.com/ACCESS-NRI/access-om3-configs/issues/257).
```

dougiesquire commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -4,90 +4,84 @@ The sections that follow explain why we selected each model parameter for the gl
-We keep most of parameters by default the same as the [GFDL OM5 configuration](https://github.com/NOAA-GFDL/MOM6-examples/blob/3c1de3512e2200bfc10d9e5150715c9df76dbd30/ice_ocean_SIS2/Baltic_OM5_025/MOM_parameter_doc.all). We incorporate Langmuir turbulence effects - `EPBL_LANGMUIR_SCHEME = “ADDITIVE”`. This choice adds another mixing contribution due to Langmuir circulations (wave-driven mixing). ...
+### Vertical mixing parameterisations
+#### Energetic planetary boundary layer (ePBL)
+The configuration handles the vertical mixing in the ocean surface boundary layer with the ePBL scheme rather the the traditional KPP. The ePBL scheme is an energy-based 1D turbulence closure approach that integrates a boundary layer energy budget to determine mixing coefficients. It was developed by [@reichl2018simplified] to improve upon KPP for climate simulations by including the effect of tu ...
```
```suggestion
The configuration handles the vertical mixing in the ocean surface boundary layer with the ePBL scheme rather the the traditional KPP. The ePBL scheme is an energy-based 1D turbulence closure approach that integrates a boundary layer energy budget to determine mixing coefficients. It was developed by [@reichl2018simplified] to improve upon KPP for climate simulations by including the effect of turbulent kinetic energy input and wind-driven mixing in a more physically constrained way. Relevant discussion can be found in [issues/465](https://github.com/ACCESS-NRI/access-om3-configs/issues/465), [issues/426](https://github.com/ACCESS-NRI/access-om3-configs/issues/426), [issues/373](https://github.com/ACCESS-NRI/access-om3-configs/issues/373).
```

dougiesquire commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -4,90 +4,88 @@ The sections that follow explain why we selected each model parameter for the gl
-2. $\kappa_{\text{Smagorinsky}}$ (`SMAGORINSKY_KH = False`) is from the dynamic Smagorinsky scheme,
-3. $\kappa_{\text{Leith}}$ (`LEITH_KH = False`) is the Leith viscosity.
+- no constant background viscosity (`AH = 0.0`)
+- a grid-dependent background viscosity (`AH_TIME_SCALE = 0.0`, `AH_TIME_SCALE = 0.0`)
```
```suggestion
- a grid-dependent background viscosity (`AH_VEL_SCALE = 0.01`, `AH_TIME_SCALE = 0.0`)
```

chrisb13 commented:
Suggestions (still editing)

- [ ] "Use this as both a quick reference and a roadmap for deeper dives into the individual configuration files such as [MOM_parameter_doc.all](https://github.com/ACCESS-NRI/access-om3-configs/blob/release-MC_25km_jra_ryf/docs/MOM_parameter_doc.all)" this bit should reference the other `iaf` doc as well or change the wording.
- [ ] Suggest some re-phrasing, something like: "deeper dives into the individual configuration files such as [MOM_parameter_doc.all](https://github.com/ACCESS-NRI/access-om3-configs/blob/release-MC_25km_jra_ryf/docs/MOM_parameter_doc.all) **which is** the list of all [MOM6](https://github.com/ACCESS-NRI/mom6) parameters. **In the context of CICE:** [cice_in](https://github.com/ACCESS-NRI/access-om3-configs/blob/release-MC_25km_jra_ryf/ice_in), the [CICE](https://github.com/ACCESS-NRI/cice) namelist file."

chrisb13 commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -4,90 +4,88 @@ The sections that follow explain why we selected each model parameter for the gl
 
-The 25km configuration uses a tripolar grid to avoid a singularity at the North Pole. The domain is zonally periodic `REENTRANT_X = True` and open at the north via a tripolar fold `TRIPOLAR_N = True` while closed in the south `REENTRANT_Y = False`. The horizontal grid has `1440x1152` tracer points. This is closely aligned with prior models, such as `ACCESS-OM2-025` and `GFDL OM4/OM5` (`1440x1080` ...
+### Grid resolution
+The configuration uses a zonally-periodic tripolar grid to avoid a singularity at the North Pole (`REENTRANT_X = True`, `TRIPOLAR_N = True`). The horizontal grid has 1440 x 1152 tracer points providing eddy-permitting resolution similar to ACCESS-OM2-025 and GFDL's OM4/OM5 global models (1440 x 1080). See [Grids](../../inputs/Grids/) for more information. 
```
On this bit
> GFDL's OM4/OM5 global models (1440 x 1080)

Can we have a link for that?

minghangli-uni replied:
Here's the link https://github.com/NOAA-GFDL/MOM6-examples/blob/f33849827fba879cf0a2ef39aba11822daef93f1/ice_ocean_SIS2/OM4_025/MOM_parameter_doc.all#L219-L224

chrisb13 commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -4,90 +4,88 @@ The sections that follow explain why we selected each model parameter for the gl
 
-The freezing conservative temperature is calculated from absolute salinity and pressure using a 23-term polynomial fit refactored from the TEOS-10 package (`TFREEZE_FORM = "TEOS_POLY"`). More relevant discussions or notes can be found in [TWG-23-July-2025](https://forum.access-hive.org.au/t/cosima-twg-meeting-minutes-2025/4067/17#:~:text=Freezing%20temperature%20consistency%20between%20mom6%20and ...
+### Thermodynamics and Equation of State (TEOS-10)
+The configuration uses the "ROQUET_RHO" equation of state for seawater thermodynamic properties (`EQN_OF_STATE = "ROQUET_RHO"`). This scheme is based on TEOS-10, but uses a 75-term polynomial to compute in-situ density as a function of conservative temperature and absolute salinity, closely approximating the full TEOS-10 results [@roquet2015accurate]. The `_RHO` variant specifically fits density  ...
```
```suggestion
The configuration uses the "ROQUET_RHO" equation of state for seawater thermodynamic properties (`EQN_OF_STATE = "ROQUET_RHO"`). This scheme is based on TEOS-10 using a 75-term polynomial to compute in-situ density as a function of conservative temperature and absolute salinity, closely approximating the full TEOS-10 results [@roquet2015accurate]. The `_RHO` variant specifically fits density rather than specific volume and is well-suited for layered models. Prognostic temperature and salinity are conservative temperature and absolute salinity (`USE_CONTEMP_ABSSAL = True`), consistent with the equation of state.
```

chrisb13 replied:
Currently we have:
https://access-om3-configs.access-hive.org.au/pr-preview-912/inputs/Initial-conditions-generation/

> The workflow produces Conservative Temperature (CT) and Absolute Salinity (SA), the prognostic variables required by the TEOS-10 equation of state (EOS) for use in ACCESS-OM3.

I think it would be worth mentioning or linking to that here.

For consistency, would also be re-wording

chrisb13 commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -4,90 +4,88 @@ The sections that follow explain why we selected each model parameter for the gl
 
-### Vertical resolution and ALE coordinate
-This configuration uses 75 vertical layers (`NK=75`) with an arbitrary Lagrangian Euler (`ALE`) vertical coordinate scheme [@griffies2020primer]. The `ALE` scheme is enabled by `USE_REGRIDDING = True` (activating the “split-explicit” layered/regridding algorithm). MOM6 also supports true hybrid vertical coordinates, such as "layered isopycnal-z", where layers follow density surfaces in the ocean  ...
+There are 75 vertical layers (`NK = 75`). MOM6 supports an arbitrary Lagrangian Euler (ALE) vertical coordinate scheme [@griffies2020primer] that allows for completely general vertical coordinates. A number of coordinates are supported "out-of-the-box", including isopycnal, geopotential, terrain-following, HYCOM1 etc. This configuration uses ALE with a stretched geopotential (z-star) vertical coo ...
```
Thanks @ezhilsabareesh8. Can I suggest you add that detail?

I note that [currently we have](https://access-om3-configs.access-hive.org.au/inputs/Grids/#vertical-grid):
> Vertical grid
In the ocean model, we use a [75 level vertical grid](https://github.com/COSIMA/om3-scripts/blob/main/grid_generation/generate_vertical_grid.py) unchanged from many OM2 configurations, following Stewart et al.(2017)(Stewart et al., 2017)[2](https://access-om3-configs.access-hive.org.au/inputs/Grids/#fn:StewartHoggGriffiesHeerdegenWardSpenceEngland2017a).

I think putting it here^ makes more sense? Then just adding a link from here. If you do want to add it [here](https://access-om3-configs.access-hive.org.au/inputs/Grids/#vertical-grid), please do it on this PR:
https://github.com/ACCESS-NRI/access-om3-configs/pull/912/
(Just add your commit to the branch.)

Is it possible to even list what the layers are (as in what they start at)?

@dougiesquire has just been looking at how this varies with z* -- a comment in that sense might be nice too but not necessary.

chrisb13 replied:
Thanks @ezhilsabareesh8 can you please add a reference to that here too? "Further details of the vertical grid can be found..." or similar.

chrisb13 commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -4,90 +4,88 @@ The sections that follow explain why we selected each model parameter for the gl
 
-### Vertical resolution and ALE coordinate
-This configuration uses 75 vertical layers (`NK=75`) with an arbitrary Lagrangian Euler (`ALE`) vertical coordinate scheme [@griffies2020primer]. The `ALE` scheme is enabled by `USE_REGRIDDING = True` (activating the “split-explicit” layered/regridding algorithm). MOM6 also supports true hybrid vertical coordinates, such as "layered isopycnal-z", where layers follow density surfaces in the ocean  ...
+There are 75 vertical layers (`NK = 75`). MOM6 supports an arbitrary Lagrangian Euler (ALE) vertical coordinate scheme [@griffies2020primer] that allows for completely general vertical coordinates. A number of coordinates are supported "out-of-the-box", including isopycnal, geopotential, terrain-following, HYCOM1 etc. This configuration uses ALE with a stretched geopotential (z-star) vertical coo ...
```
```suggestion
There are 75 vertical layers (`NK = 75`). MOM6 supports an arbitrary Lagrangian Euler (ALE) vertical coordinate scheme [@griffies2020primer] that allows for completely general vertical coordinates. A number of coordinates are supported "out-of-the-box", including isopycnal, geopotential, terrain-following, HYCOM1 etc. This configuration uses ALE with a stretched geopotential (z-star) vertical coordinate (`USE_REGRIDDING = True`, `REGRIDDING_COORDINATE_MODE = "ZSTAR"`). The layer spacing specified via an input file (`ALE_COORDINATE_CONFIG = "FILE:ocean_vgrid.nc,interfaces=zeta"`). The deepest ocean depth is set to `MAXIMUM_DEPTH = 6000.0`. Further details of the vertical grid can be found [here](../../inputs/Grids/#vertical-grid).
```

chrisb13 commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -4,90 +4,88 @@ The sections that follow explain why we selected each model parameter for the gl
-### Diagnostics and age tracer
-The configuration introduces some passive tracers and diagnostics for analysis. For example, we enable `USE_IDEAL_AGE_TRACER = True`, which measures the time since water left the surface. This tracer ages at a rate of 1/year once it is isolated from the surface (`DO_IDEAL_AGE = True`). It doesn’t affect dynamics but is a diagnostic to understand water mass ventilation and residence times.
+### Diagnostics
+Three-dimensional ocean diagnostics are output on either $z*$- or density-coordinates, depending on the diagnostic, rather than on the model's native coordinate. Specifically, `NUM_DIAG_COORDS = 2` with `DIAG_COORDS = "z Z ZSTAR", "rho2 RHO2 RHO"` and the vertical coordinate levels for each are defined by `DIAG_COORD_DEF_Z = "FILE:ocean_vgrid.nc,interfaces=zeta"` and `DIAG_COORD_DEF_RHO2 = "RFNC1 ...
```
Would it be possible to give a link(s) or some information as to which variables are output on which coordinates?

dougiesquire replied:
We can add a link to the section on [ocean diagnostics](https://github.com/ACCESS-NRI/access-om3-configs/pull/1020) once it is merged

chrisb13 commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -4,90 +4,88 @@ The sections that follow explain why we selected each model parameter for the gl
+The ePBL scheme parameters in the configuration are based on the [GFDL OM5 configuration](https://github.com/NOAA-GFDL/MOM6-examples/blob/3c1de3512e2200bfc10d9e5150715c9df76dbd30/ice_ocean_SIS2/Baltic_OM5_025/MOM_parameter_doc.all), including:
 
-We have adjusted some `ePBL` parameters to match the [`GFDL` OM4 scheme](https://github.com/NOAA-GFDL/MOM6-examples/blob/3c1de3512e2200bfc10d9e5150715c9df76dbd30/ice_ocean_SIS2/Baltic_OM5_025/MOM_parameter_doc.all#L2247) (`EPBL_MSTAR_SCHEME = “OM4”`). We set `MSTAR_CAP = 1.25` (caps the mixing length scale factor `m` to 1.25) and adjusted coefficients: `MSTAR2_COEF1 = 0.29` and `COEF2 = 0.152`. T ...
+- Additional mixing due to Langmuir (wave-driven) turbulence (`EPBL_LANGMUIR_SCHEME = “ADDITIVE”`). Since we do not explicitly couple to a wave model in this configuration (`USE_WAVES = False`), the Langmuir effect is parameterised via a predetermined enhancement factor using MOM6 default values. The inclusion of wave effects is expected to reduce warm SST biases by enhancing mixing under strong  ...
```
> The inclusion of wave effects is expected to reduce warm SST biases by enhancing mixing under strong winds

is that what we've found in `mcw` @ezhilsabareesh8 @NoahDay ?

ezhilsabareesh8 replied:
When WW3 is fully coupled to ePBL vertical mixing, we find that the resulting mixed-layer depths are unrealistic; therefore, we disable wave-induced mixing in ePBL (i.e., no wave to ocean coupling) to obtain reasonable mixed-layer depths.

anton-seaice commented:
The link changed during this PR, its lost in the comments above.

Try 

https://access-om3-configs.access-hive.org.au/pr-previews/911/

Not sure if youll also need to merge with latest main

dougiesquire commented:
> (fyi @dougiesquire I notice there's some `MC_25KM.md` conflicts too)

Yeah, this is due to the merging of [this PR](https://github.com/ACCESS-NRI/access-om3-configs/pull/928). I'll fix up when addressing your comments.

minghangli-uni commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -4,92 +4,111 @@ The sections that follow explain why we selected each model parameter for the gl
+- isopycnal tracer diffusivity ($KHTR$ in MOM6)
 
-We enable `BOUND_KH = True` to locally limit the Laplacian diffusivity ensuring CFL stability. Specifically, a coefficient `Kh_Limit = 0.3 / (dt * 4.0)` is applied, taking grid spacing into account. To further improve numerical stability, we enable both `BETTER_BOUND_KH = True` and `BETTER_BOUND_AH = True`, which apply more refined constraints on Laplacian and biharmonic viscosities, respectively ...
+In MOM5, these coefficients were prescribed constants or latitude-dependent maps, but these choices are ad-hoc and not dynamically constrained. Hence MOM6 also offers the Mesoscale Eddy Kinetic Energy (MEKE) scheme [@mom6_mom_meke] which provides a flow-dependent, scale-aware GM diffusivity. MEKE prognostically computes an eddy kinetic energy field $E(x,y,z,t)$ from which it derives an eddy veloc ...
```
```suggestion
In MOM5, these coefficients were prescribed constants or latitude-dependent maps, but these choices are ad-hoc and not dynamically constrained. Hence MOM6 also offers the Mesoscale Eddy Kinetic Energy (`MEKE`) scheme [@mom6_mom_meke] which provides a flow-dependent, scale-aware `GM` diffusivity. `MEKE` prognostically computes an eddy kinetic energy field $E(x,y,z,t)$ from which it derives an eddy velocity scale,
```

minghangli-uni commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -4,92 +4,111 @@ The sections that follow explain why we selected each model parameter for the gl
+KHTH = CU_{eddy}L
+$$,
+
+Hence when EKE is large, GM diffusivity increases with stronger flattening of isopycnals and similarly when EKE is small, GM diffusivity decreases and let the resolved eddies do the work. Hence the GM flattening of isopycnals becomes energetically consistent and scale-aware, adapting automatically to the local eddy field.
```
```suggestion
Hence when EKE is large, `GM` diffusivity increases with stronger flattening of isopycnals and similarly when EKE is small, `GM` diffusivity decreases and let the resolved eddies do the work. Hence the `GM` flattening of isopycnals becomes energetically consistent and scale-aware, adapting automatically to the local eddy field.
```

minghangli-uni commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -4,92 +4,111 @@ The sections that follow explain why we selected each model parameter for the gl
+This configuration uses MEKE (`USE_MEKE = True`) to provide the GM diffusivity. Because we do not supply an external EKE field (`EKE_SOURCE = "prog"`), EKE is generated internally through instability growth. `MEKE_BGSRC = 1.0E-13` prevents `EKE` from decaying to zero in very quiet regions. It serves as a floor to aid numerical stability and is analogous to a background diffusivity but in energy f ...
 
-By using `MEKE`, the model is effectively resolution-aware, as resolution increases and resolves more eddies, the diagnostic EKE and hence `GM` coefficient naturally reduces. At the same time , in coarser areas or higher latitudes where eddies are still under-resolved, `MEKE` ramps up the eddy mixing. This avoids the need for ad-hoc spatial maps of `GM` coefficients. By using `FGNV`[@ferrari2010b ...
+With `MEKE`, MOM6 becomes explicitly resolution-aware: as horizontal resolution increases and more eddy processes are partially resolved, the diagnosed GM coefficient naturally decreases; conversely, GM strengthens where eddies remain under-resolved (e.g. high latitudes), removing the need for ad-hoc spatial maps of GM coefficients.
```
```suggestion
With `MEKE`, MOM6 becomes explicitly resolution-aware: as horizontal resolution increases and more eddy processes are partially resolved, the diagnosed `GM` coefficient naturally decreases; conversely, `GM` strengthens where eddies remain under-resolved (e.g. high latitudes), removing the need for ad-hoc spatial maps of `GM` coefficients.
```

minghangli-uni commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -4,92 +4,111 @@ The sections that follow explain why we selected each model parameter for the gl
 
-The full viscosity includes the dynamic components,
+### Isopycnal mixing
+Baroclinic instability converts available potential energy (APE) stored in sloping isopycnals into eddy kinetic energy (EKE). In an eddy-permitting 25km configuration, this conversion is only partly resolved hence the model does not fully capture the eddy processes that naturally flatten isopycnals and release APE. As a result, isopycnal slopes remain steeper than they should be unless the unreso ...
```
Sorry for the confusion. I meant details of the physical basis and formulation of GM. 

```suggestion
Baroclinic instability converts available potential energy (APE) stored in sloping isopycnals into eddy kinetic energy (EKE). In an eddy-permitting 25km configuration, this conversion is only partly resolved hence the model does not fully capture the eddy processes that naturally flatten isopycnals and release APE. As a result, isopycnal slopes remain steeper than they should be unless the unresolved eddy effects are parameterised. More details on the physical basis and mathematical formulation of the GM parameterisation can be found in [@mitgcm_gmredi].
```

dougiesquire commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -4,92 +4,111 @@ The sections that follow explain why we selected each model parameter for the gl
+and an eddy length scale ($L$) based on a configurable combination of multiple length scales. The GM thickness diffusivity is then computed as,
+
+$$
+KHTH = CU_{eddy}L
```
```suggestion
KHTH = CU_{e}L
```

dougiesquire commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -4,92 +4,118 @@ The sections that follow explain why we selected each model parameter for the gl
+### Frazil formation
+Frazil ice typically consists of individual particles that appear like a coarse slush. Frazil ice can contribute to the creation of first-year sea ice, evolving to form a soupy ‘grease ice’ layer on the ocean surface, which may then develop into pancake ice. Frazil formation in the ocean is turned on (`FRAZIL = TRUE`, `ENABLE_THERMODYNAMICS = True`), Frazil ice forms in the model when the in situ ...
+ 
+ 1. `MASK_SRESTORE_UNDER_ICE = False`, i.e. disables SSS restoring under sea-ice based on a frazil criteria;
```
```suggestion
 1. `MASK_SRESTORE_UNDER_ICE = False`, i.e. do not disable SSS restoring under sea-ice based on a frazil criteria;
```

dougiesquire commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -4,92 +4,118 @@ The sections that follow explain why we selected each model parameter for the gl
 
 #### Interior shear-driven mixing
-Below the surface layer, we use a parameterisation for shear-driven mixing in stratified interior. Specifically we enable the [@jackson2008parameterization] shear instability scheme (`USE_JACKSON_PARAM = True`). This scheme targets mixing in stratified shear zones. It uses a local Richardson number (`Ri`). We keep the default critical Richardson number `RINO_CRIT = 0.25` and the nondimensional sh ...
+Shear-driven mixing is parameterised using the Jackson-Hallberg-Legg shear mixing scheme [@jackson2008parameterization] with the MOM6 default critical Richardson number (Ri; `RINO_CRIT = 0.25`) and shear mixing rate (`USE_JACKSON_PARAM = True` and `SHEARMIX_RATE = 0.089`). This scheme targets mixing in stratified shear zones, effectively adding interior vertical diffusivity when the local Ri is l ...
```
```suggestion
Shear-driven mixing is parameterised using the Jackson-Hallberg-Legg shear mixing scheme [@jackson2008parameterization] with the MOM6 default critical Richardson number (Ri) and shear mixing rate (`USE_JACKSON_PARAM = True`, `RINO_CRIT = 0.25` and `SHEARMIX_RATE = 0.089`). This scheme targets mixing in stratified shear zones, effectively adding interior vertical diffusivity when the local Ri is less than the critical value. The shear is computed at cell vertices to better capture shear between adjacent grid cells (`VERTEX_SHEAR = True`). The Jackson-Hallberg-Legg parameterisation is energetically constrained: it iteratively finds a diffusivity such that the energy extracted from the mean flow equals the energy used in mixing plus that lost to dissipation (`MAX_RINO_IT = 25`, inherited from the [GFDL OM5 configuration](https://github.com/NOAA-GFDL/MOM6-examples/blob/3c1de3512e2200bfc10d9e5150715c9df76dbd30/ice_ocean_SIS2/Baltic_OM5_025/MOM_parameter_doc.all#L2088)). See the [MOM6 documentation for further details](https://mom6.readthedocs.io/en/main/api/generated/pages/Internal_Vert_Mixing.html#shear-driven-mixing-in-jackson).
```

dougiesquire commented:
For a few comments/questions, it's unclear to me whether any action is wanted _in this PR_. It would be helpful if you could clarify @chrisb13, @minghangli-uni:

- https://github.com/ACCESS-NRI/access-om3-configs/pull/911#discussion_r2756877775
- https://github.com/ACCESS-NRI/access-om3-configs/pull/911#discussion_r2756885091
- https://github.com/ACCESS-NRI/access-om3-configs/pull/911#discussion_r2756884641
- https://github.com/ACCESS-NRI/access-om3-configs/pull/911#discussion_r2757139285
- https://github.com/ACCESS-NRI/access-om3-configs/pull/911#discussion_r2757139628

Other than those, I think all suggestions are now resolved.

Also [this PR](https://github.com/ACCESS-NRI/access-om3-configs/pull/912) into this branch is still outstanding. Once that is merged and the list of comments above are resolved, we'll be ready for a re-review from all of us.

minghangli-uni commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -4,92 +4,111 @@ The sections that follow explain why we selected each model parameter for the gl
+Mesoscale eddies also induce irreversible mixing of tracers but primarily along neutral density surfaces rather than vertically. This diabatic component needs to be represented through an isopycnal diffusion parameterisation that diffuses tracers along neutral surfaces while avoiding spurious diapycnal mixing.
 
-where,
+To represent both the flattening of isopycnals and the along-isopycnal tracer mixing, the configuration applies a hybrid mesoscale parameterisation that combines neutral diffusion [@redi1982oceanic] with the Gent-McWilliams (`GM`) eddy-induced advection scheme [@gent1990isopycnal]. Both require an eddy diffusivity,
```
```suggestion
To represent both the quasi-adiabatic flattening of isopycnals and along-isopycnal tracer mixing, the configuration applies a hybrid mesoscale parameterisation that combines neutral diffusion [@redi1982oceanic] with the Gent-McWilliams (`GM`) eddy-induced advection scheme [@gent1990isopycnal]. Both schemes require an eddy diffusivity,
```

minghangli-uni commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -4,92 +4,111 @@ The sections that follow explain why we selected each model parameter for the gl
 
-#### Isopycnal thickness diffusion (`GM`)
-`GM` is turned on via `THICKNESSDIFFUSE = True`. Instead of using a fixed `GM` thickness diffusivity (`KHTH = 0.0`), the Mesoscale Eddy Kinetic Energy (MEKE) scheme (`USE_MEKE = True`) is turned on. MEKE activates a prognostic equation for eddy kinetic energy (EKE) and a spatially varying GM streamfunction. The MEKE parameterisation is based on the work of [@jansen2015parameterization], where an  ...
+This configuration uses `MEKE` (`USE_MEKE = True`) to provide the `GM` diffusivity. Because we do not supply an external EKE field (`EKE_SOURCE = "prog"`), EKE is generated internally through instability growth. `MEKE_BGSRC = 1.0E-13` prevents `EKE` from decaying to zero in very quiet regions. It serves as a floor to aid numerical stability and is analogous to a background diffusivity but in ener ...
```
No action required

minghangli-uni commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -4,92 +4,111 @@ The sections that follow explain why we selected each model parameter for the gl
+and an eddy length scale ($L$) based on a configurable combination of multiple length scales. The GM thickness diffusivity is then computed as,
+
+$$
+KHTH = CU_{eddy}L
```
Done in [fffa729](https://github.com/ACCESS-NRI/access-om3-configs/pull/911/commits/fffa72998ad043d1bcb8db5b473521ddd3a5d4a8)

minghangli-uni commented:
> Thanks @minghangli-uni. Just to clarify: I think the plan is once this https://github.com/ACCESS-NRI/access-om3-configs/pull/912 is merged, we'll move the section to a new PR (https://github.com/ACCESS-NRI/access-om3-configs/pull/911#issuecomment-3808172347) (i.e. Isopycnal mixing section into a new branch, delete it from this branch and merge). That ok?

Ah yes, that's what I meant - sorry for the confusion. Looks like we'll need another round of review.

minghangli-uni commented:
Apologies if I've missed something in this long PR. I've noticed that some formulas are not rendering correctly such as \(kg/m^3\), I thought this issue had already been addressed in https://github.com/ACCESS-NRI/access-om3-configs/issues/666 so perhaps we can follow this up in a separate PR.

atteggiani commented:
A

> > Are we talking about two different local previews of the same exact code?
> 
> No, online previews.

Are you using two different browsers? Have you tried cleaning your browser cache (or opening the preview in an incognito window)?

minghangli-uni commented:
> You're on a mac, right @minghangli-uni?

Yeah I'm on macOS but a lot of us are too...

atteggiani commented:
> https://access-om3-configs.access-hive.org.au/pr-previews/911/configurations/MC_25km/

@minghangli-uni are you using the link above or using the RTD one? (Because in [this message](https://github.com/ACCESS-NRI/access-om3-configs/pull/911#issuecomment-3845753186) the URL is the readthedocs one)

chrisb13 commented:
> @chrisb13 did you want to push a change for [this](https://github.com/ACCESS-NRI/access-om3-configs/pull/912#discussion_r2766495602)? I suggest just making the change on this branch and we can review it when we all do our once-over.

Done: 
https://github.com/ACCESS-NRI/access-om3-configs/pull/911/changes#diff-f0797ef89399416ef9f53c36c6ac086d4e898f984bbcff88066534dbc7ef28d7

Feel free to edit etc.

dougiesquire commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -1,95 +1,90 @@
+ 1. `MASK_SRESTORE_UNDER_ICE = False`, i.e. do not disable SSS restoring under sea-ice based on a frazil criteria;
+  1. `PRESSURE_DEPENDENT_FRAZIL = True` a pressure dependent freezing temperature is used when making frazil;
+  1. `RECLAIM_FRAZIL = True` use any frazil heat deficit to cool any overlying layers down to the freezing point.
 
```
```suggestion
Frazil ice typically consists of individual particles that appear like a coarse slush. Frazil ice can contribute to the creation of first-year sea ice, evolving to form a soupy ‘grease ice’ layer on the ocean surface, which may then develop into pancake ice. Frazil formation in the ocean is turned on (`FRAZIL = TRUE`). Frazil ice forms in the model when the in situ temperature drops below the local freezing point, taking into account the in situ salinity and pressure (`PRESSURE_DEPENDENT_FRAZIL = True`). The scheme works upwards through each water column, transferring heat downwards from the layer above as needed to prevent the in-situ temperature falling below the local freezing point in each layer in turn. If the top layer is below freezing, heat is extracted from the sea ice model, which grows frazil ice in response. In these configurations, we also have:
 
 1. `MASK_SRESTORE_UNDER_ICE = False`, i.e. do not disable SSS restoring under sea-ice based on a frazil criteria;
  1. `RECLAIM_FRAZIL = True` use any frazil heat deficit to cool any overlying layers down to the freezing point.

```

dougiesquire commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -1,95 +1,90 @@
+  1. `RECLAIM_FRAZIL = True` use any frazil heat deficit to cool any overlying layers down to the freezing point.
 
-We ensure salinity never goes negative by setting `BOUND_SALINITY = True`. In coupled models, sea-ice formation and melting can generate large salinity fluxes at the ocean surface. This setting clips salinity at a minimum of `MIN_SALINITY = 0.0`. However, if the lower bound is hit, it clips the salinity value and discards any excess, which may violate salt conservation in rare cases. This can occ ...
+More details are [here](https://mom6.readthedocs.io/en/main/api/generated/pages/Frazil_Ice.html.
```
```suggestion
More details are [here](https://mom6.readthedocs.io/en/main/api/generated/pages/Frazil_Ice.html).
```

dougiesquire commented on documentation/docs/pages/configurations/MC_25km.md:
```diff
@@ -1,95 +1,90 @@
 
 #### Interior shear-driven mixing
-Below the surface layer, we use a parameterisation for shear-driven mixing in stratified interior. Specifically we enable the [@jackson2008parameterization] shear instability scheme (`USE_JACKSON_PARAM = True`). This scheme targets mixing in stratified shear zones. It uses a local Richardson number (`Ri`). We keep the default critical Richardson number `RINO_CRIT = 0.25` and the nondimensional sh ...
+Shear-driven mixing is parameterised using the Jackson-Hallberg-Legg shear mixing scheme [@jackson2008parameterization] with the MOM6 default critical Richardson number (Ri) and shear mixing rate (`USE_JACKSON_PARAM = True`, `RINO_CRIT = 0.25` and `SHEARMIX_RATE = 0.089`). This scheme targets mixing in stratified shear zones, effectively adding interior vertical diffusivity when the local Ri is l ...
```
```suggestion
Shear-driven mixing is parameterised using the Jackson-Hallberg-Legg shear mixing scheme [@jackson2008parameterization] with the MOM6 default critical Richardson number (Ri) and shear mixing rate (`USE_JACKSON_PARAM = True`, `RINO_CRIT = 0.25` and `SHEARMIX_RATE = 0.089`). This scheme targets mixing in stratified shear zones, effectively adding interior vertical diffusivity when the local Ri is less than the critical value. The shear is computed at cell vertices to better capture shear between adjacent grid cells (`VERTEX_SHEAR = True`). The Jackson-Hallberg-Legg parameterisation is energetically constrained: it iteratively finds a diffusivity such that the energy extracted from the mean flow equals the energy used in mixing plus that lost to dissipation (`MAX_RINO_IT = 25`; inherited from the [GFDL OM5 configuration](https://github.com/NOAA-GFDL/MOM6-examples/blob/3c1de3512e2200bfc10d9e5150715c9df76dbd30/ice_ocean_SIS2/Baltic_OM5_025/MOM_parameter_doc.all#L2088)). See the [MOM6 documentation for further details](https://mom6.readthedocs.io/en/main/api/generated/pages/Internal_Vert_Mixing.html#shear-driven-mixing-in-jackson).
```

chrisb13 commented:
> Probably after the isopycnal section? [#911 (comment)](https://github.com/ACCESS-NRI/access-om3-configs/pull/911#issuecomment-3808172347)

I don't think I'm following what you mean here. I think you deleted that section as discussed? Have you opened a new PR with the updated text?

chrisb13 commented:
Ok, will do but let's just wait to hear back from @minghangli-uni on [this point](https://github.com/ACCESS-NRI/access-om3-configs/pull/911#issuecomment-3858007469) -- make sure we're all on the same plan.

--

Oh he just replied!
